packages/client/wallets/walletconnect/src/utils/walletconnect/mockRequiredNamespaceMethods.ts

```diff
@@ -0,0 +1,29 @@
+import { ProposalTypes } from "@walletconnect/types";
+import { BuildApprovedNamespacesParams } from "@walletconnect/utils";
+
+export function mockRequiredNamespaceMethods(
+    requiredNamespaces: ProposalTypes.RequiredNamespaces,
+    supportedNamespaces: BuildApprovedNamespacesParams["supportedNamespaces"]
+) {
+    const mockedNamespaces = { ...supportedNamespaces };
+
+    Object.entries(requiredNamespaces).forEach(([caipKey, requiredNamespace]) => {
+        if (!mockedNamespaces[caipKey]) {
+            return;
+        }
+
+        const unsupportedRequiredMethods = requiredNamespace.methods.filter(
+            (method) => !mockedNamespaces[caipKey].methods?.includes(method)
+        );
+        console.warn(
+            "[mockRequiredNamespaceMethods] Mocking unsupported required methods:",
```

Comment:
As a dev could see this, I suggest rewording to be more clear to:

`Mocking unsupported required methods: ${methods}. Calling these methods will result in a no-op.`